source/channel.c

```diff
@@ -506,8 +507,24 @@ bool aws_channel_thread_is_callers_thread(struct aws_channel *channel) {
     return aws_event_loop_thread_is_callers_thread(channel->loop);
 }
 
+static void s_update_channel_slot_message_overheads(struct aws_channel *channel) {
+    size_t overhead = 0;
+    struct aws_channel_slot *slot_iter = channel->first;
+    while (slot_iter) {
+        slot_iter->upstream_message_overhead = overhead;
+
+        if (slot_iter->handler) {
```

Comment:
If you make this `if (slot_iter->handler && slot_iter->handler->vtable->message_overhead)` this function will become optional if you have 0 overhead.